dspy/teleprompt/mipro_optimizer_v2.py

```diff
@@ -254,9 +254,9 @@ def compile(
             evaluate = Evaluate(devset=trainset, metric=self.metric, **eval_kwargs)
 
             # Determine the number of fewshot examples to use to generate demos for prompt
-            if max_bootstrapped_demos == 0 and max_labeled_demos == 0:
+            if max_bootstrapped_demos == 0:
                 max_bootstrapped_demos_for_candidate_gen = BOOTSTRAPPED_FEWSHOT_EXAMPLES_IN_CONTEXT
```

Comment:
shouldn't this be set to 0 to ensure bootstrapped_demos in the few-shot generation is 0 as intended to only use labeled demos?

Reply:
In the current implementation of MIPRO and MIPROv2 at least one bootstrapped demo is required for the candidate generation of the meta-prompt. By setting it like this, bootstrapped demos are only used for the meta-prompt but not in the final prompt.